packages/utils/src/instrument.ts

```diff
@@ -357,179 +357,228 @@ function instrumentHistory(): void {
-    ) => () => void {
-      return function(
-        this: any,
-        eventName: string,
-        fn: EventListenerOrEventListenerObject,
-        options?: boolean | EventListenerOptions,
-      ): () => void {
-        try {
-          original.call(this, eventName, ((fn as unknown) as WrappedFunction).__sentry_wrapped__, options);
-        } catch (e) {
-          // ignore, accessing __sentry_wrapped__ will throw in some Selenium environments
-        }
-        return original.call(this, eventName, fn, options);
-      };
-    });
-  });
+  return true;
 }
 
-const debounceDuration: number = 1000;
-let debounceTimer: number = 0;
-let keypressTimeout: number | undefined;
-let lastCapturedEvent: Event | undefined;
-
 /**
  * Wraps addEventListener to capture UI breadcrumbs
- * @param name the event name (e.g. "click")
  * @param handler function that will be triggered
- * @param debounce decides whether it should wait till another event loop
+ * @param globalListener indicates whether event was captured by the global event listener
  * @returns wrapped breadcrumb events handler
  * @hidden
  */
-function domEventHandler(name: string, handler: Function, debounce: boolean = false): (event: Event) => void {
+function makeDOMEventHandler(handler: Function, globalListener: boolean = false): (event: Event) => void {
   return (event: Event): void => {
-    // reset keypress timeout; e.g. triggering a 'click' after
-    // a 'keypress' will reset the keypress debounce so that a new
-    // set of keypresses can be recorded
-    keypressTimeout = undefined;
     // It's possible this handler might trigger multiple times for the same
-    // event (e.g. event propagation through node ancestors). Ignore if we've
-    // already captured the event.
+    // event (e.g. event propagation through node ancestors).
+    // Ignore if we've already captured that event.
     if (!event || lastCapturedEvent === event) {
       return;
     }
 
-    lastCapturedEvent = event;
+    // We always want to skip _some_ events.
+    if (shouldSkipDOMEvent(event)) {
+      return;
+    }
+
+    const name = event.type === 'keypress' ? 'input' : event.type;
 
-    if (debounceTimer) {
-      clearTimeout(debounceTimer);
+    // If there is no debounce timer, it means that we can safely capture the new event and store it for future comparisons.
+    if (debounceTimerID === undefined) {
+      handler({
+        event: event,
+        name,
+        global: globalListener,
+      });
+      lastCapturedEvent = event;
     }
 
-    if (debounce) {
-      debounceTimer = setTimeout(() => {
-        handler({ event, name });
+    // If there is a debounce awaiting, see if the new event is different enough to treat it as a unique one.
+    // If that's the case, emit the instrumentation event and store locally newly captured DOM event.
+    if (debounceTimerID !== undefined && shouldShortcircuitPreviousDebounce(lastCapturedEvent, event)) {
```

Comment:
Is there a reason this is a separate `if` rather than a `||` on the previous condition? Would we ever want both things to happen (if, say, a timer got set between the first `if` and the second `if`)? IDK if that's even possible, but in case it did happen, would we actually want to capture the breadcrumb twice?